pkg/kv/dynamodb/store.go

```diff
@@ -455,3 +469,48 @@ func (e *EntriesIterator) Err() error {
 func (e *EntriesIterator) Close() {
 	e.err = kv.ErrClosedEntries
 }
+
+// StartPeriodicCheck performs one check and continues every 'interval' in the background
+func (s *Store) StartPeriodicCheck() {
+	interval := s.params.HealthCheckInterval
+	if interval <= 0 || s.cancel != nil {
+		return
+	}
+	s.cancel = make(chan bool)
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		defer func() { s.cancel = nil }()
+		s.logger.WithField("interval", interval).Debug("Starting DynamoDB health check")
+		// check first and loop for checking every interval
+		s.Check()
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				s.Check()
+			case <-s.cancel:
+				return
+			}
+		}
+	}()
+}
+
+func (s *Store) Check() {
+	log := s.logger.WithField("store_type", DriverName)
+	success, err := isTableExist(context.Background(), s.svc, s.params.TableName)
+	if success {
+		log.Debug("DynamoDB health check passed!")
+	} else {
+		log.WithError(err).Debug("DynamoDB health check failed")
+	}
+}
+
+func (s *Store) StopPeriodicCheck() {
+	if s.cancel != nil {
+		s.logger.Info("Stopping DynamoDB health check")
+		s.cancel <- true
+		s.wg.Wait()
+	}
```

Comment:
To simplify - just close the channel and wait for the wg.
Log probably in debug level - not so interesting.